packages/lib/gpc/test/gpc.spec.ts

```diff
@@ -178,61 +185,141 @@ describe("gpc library (Compiled test artifacts) should work", async function ()
     }
   });
 
+  const pod1 = POD.sign(sampleEntries, privateKey);
+  const typicalProofConfig: GPCProofConfig = {
+    pods: {
+      pod1: {
+        entries: {
+          A: { isRevealed: true },
+          E: {
+            isRevealed: false,
+            equalsEntry: "pod1.A",
+            isMemberOf: "list1"
+          },
+          owner: { isRevealed: false, isOwnerID: true }
+        },
+        signerPublicKey: {
+          isRevealed: false,
+          isMemberOf: "admissiblePubKeys"
+        }
+      }
+    }
+  };
+  const typicalProofInputs: GPCProofInputs = {
+    pods: { pod1 },
+    owner: {
+      semaphoreV3: ownerIdentity,
+      externalNullifier: { type: "int", value: 42n }
+    },
+    membershipLists: {
+      list1: [sampleEntries.F, sampleEntries.E],
+      admissiblePubKeys: [
+        pod1.signerPublicKey,
+        "f71b62538fbc40df0d5e5b2034641ae437bdbf06012779590099456cf25b5f8f",
+        "755224af31d5b5e47cc6ca8827b8bf9d2ceba48bf439907abaade0a3269d561b",
+        "f27205e5ceeaad24025652cc9f6f18cee5897266f8c0aac5b702d48e0dea3585",
+        "2af47e1aaf8f0450b9fb2e429042708ec7d173c4ac4329747db1063b78db4e0d"
+      ].map(PODEdDSAPublicKeyValue)
+    },
+    watermark: { type: "int", value: 1337n }
+  };
+  const expectedRevealedClaimsForTypicalCase: GPCRevealedClaims = {
+    pods: {
+      pod1: {
+        entries: { A: { type: "int", value: 123n } }
+      }
+    },
+    owner: {
+      externalNullifier: { type: "int", value: 42n },
+      nullifierHash: poseidon2([
+        makeWatermarkSignal({ type: "int", value: 42n }),
+        ownerIdentity.nullifier
+      ])
+    },
+    membershipLists: typicalProofInputs.membershipLists,
+    watermark: { type: "int", value: 1337n }
+  };
+
   it("should prove and verify a typical case", async function () {
-    const pod1 = POD.sign(sampleEntries, privateKey);
-    const proofConfig: GPCProofConfig = {
-      pods: {
-        pod1: {
-          entries: {
-            A: { isRevealed: true },
-            E: {
-              isRevealed: false,
-              equalsEntry: "pod1.A",
-              isMemberOf: "list1"
-            },
-            owner: { isRevealed: false, isOwnerID: true }
+    const { isVerified } = await gpcProofTest(
+      typicalProofConfig,
+      typicalProofInputs,
+      expectedRevealedClaimsForTypicalCase
+    );
+    expect(isVerified).to.be.true;
+  });
+
+  it("should prove and verify a typical case with an equality check involving a non-virtual entry and a virtual entry in that order", async function () {
```

Comment:
```suggestion
  it("should prove and verify checking equality non-virtual to virtual", async function () {
```
Similar for the others.  I think the description was correct, but unweildy to read and understand